fork/fork.go

```diff
@@ -16,20 +16,34 @@
 
 package fork
 
+import (
+	"errors"
+	"math/big"
+
+	"github.com/klaytn/klaytn/params"
+)
+
 var (
-	// hardForkConfig will not be changed unless it is a test code.
-	// The test code can override this value via `UpdateHardForkConfig`.
-	hardForkConfig = &HardForkConfig{}
+	// hardForkBlockNumberConfig contains only hardFork block number
+	hardForkBlockNumberConfig *params.ChainConfig
 )
 
-// HardForkConfig defines a block number for each hard fork feature.
-type HardForkConfig struct {
+// Rules returns the hard fork information
+// CAUTIOUS: Use it when chainConfig value is not reachable
+func Rules(blockNumber *big.Int) (*params.Rules, error) {
+	if hardForkBlockNumberConfig == nil {
+		return nil, errors.New("hardForkBlockNumberConfig variable is not initialized")
+	}
+	rules := hardForkBlockNumberConfig.Rules(blockNumber)
+	return &rules, nil
 }
 
-// UpdateHardForkConfig sets values in HardForkConfig if it is not nil.
-// NOTE: this is only for test code to give flexibility of the test code.
-func UpdateHardForkConfig(h *HardForkConfig) {
-	if h != nil {
-		hardForkConfig = h
+// SetHardForkBlockNumberConfig sets values in HardForkConfig if it is not nil.
+// CAUTIOUS: Calling this function is can be dangerous, so avoid using it except tests
+func SetHardForkBlockNumberConfig(h *params.ChainConfig) error {
+	if h == nil {
+		return errors.New("hardForkBlockNumberConfig cannot be initialized as nil")
 	}
+	hardForkBlockNumberConfig = h
```

Comment:
@yoomee1313 Is it possible to call this method several times? otherwise, how about using `once.Do` method to assigne the `hardForkBlockNumberConfig`?

Reply:
@jeongkyun-oh `SetHardForkBlockNumberConfig` is only called when the node starts running. I think `once.Do` is good idea to limit the usage of this statement `hardForkBlockNumberConfig = h`. 
Applied commit: https://github.com/klaytn/klaytn/pull/1006/commits/3b23675412a2982a4be48f210b2038daaa45c69b